src/tlo/util.py

```diff
@@ -498,15 +513,21 @@ def clean_dataframe(dataframes_dict: dict[str, DataFrame]) -> None:
         for _key, dataframe in dataframes_dict.items():
             all_data[_key] = dataframe.drop(dataframe.filter(like='Unnamed'), axis=1)  # filter and drop Unnamed columns
 
-    if files is None:
-        for f_name in folder.rglob("*.csv"):
-            all_data[f_name.stem] = pd.read_csv(f_name)
-
+    return_dict = False # a flag that will determine whether the output should be a dictionary or a DatFrame
+    if isinstance(files, list):
+        return_dict = True
+    elif files is None:
+        files = [f_name.stem for f_name in folder.rglob("*.csv")]
+        return_dict = True
+    elif isinstance(files, str):
+        files = [files]
     else:
-        for f_name in files:
-            all_data[f_name] = pd.read_csv((folder / f_name).with_suffix(".csv"))
+        files = [f_name.stem for f_name in folder.rglob("*.csv")]
```

Comment:
Because of `rglob` here, if you have two files with the same name but in different sub-directories, you'll get one overwriting the other. Should we only read csv files in the given directory, and ignore subdirectories?